ast/output_test.go

```diff
@@ -0,0 +1,215 @@
+			Output: &Output{},
+			Scope: nil,
+			ReturnType: TypeString,
+		},
+		{
+			Name: "Single string expression",
+			Output: &Output{
+				Exprs: []Node{
+					&LiteralNode{
+						Value: "Whatever",
+						Typex: TypeString,
+					},
+				},
+			},
+			Scope: nil,
+			ReturnType: TypeString,
+		},
+		{
+			Name: "Single list expression of strings",
+			Output: &Output{
+				Exprs: []Node{
+					&VariableAccess{
+						Name: "testvar",
+					},
+				},
+			},
+			Scope: &BasicScope{
+				VarMap: map[string]Variable{
+					"testvar": Variable{
+						Type: TypeList,
+						Value: []Variable{
+							Variable{
+								Type: TypeString,
+								Value: "Hello",
+							},
+							Variable{
+								Type: TypeString,
+								Value: "World",
+							},
+						},
+					},
+				},
+			},
+			ReturnType: TypeList,
+		},
+		{
+			Name: "Single map expression",
+			Output: &Output{
+				Exprs: []Node{
+					&VariableAccess{
+						Name: "testvar",
+					},
+				},
+			},
+			Scope: &BasicScope{
+				VarMap: map[string]Variable{
+					"testvar": Variable{
+						Type: TypeMap,
+						Value: map[string]Variable{
+							"key1": Variable{
+								Type: TypeString,
+								Value: "Hello",
+							},
+							"key2": Variable{
+								Type: TypeString,
+								Value: "World",
+							},
+						},
+					},
+				},
+			},
+			ReturnType: TypeMap,
+		},
+		{
+			Name: "Multiple map expressions",
+			Output: &Output{
+				Exprs: []Node{
+					&VariableAccess{
+						Name: "testvar",
+					},
+					&VariableAccess{
+						Name: "testvar",
+					},
+				},
+			},
+			Scope: &BasicScope{
+				VarMap: map[string]Variable{
+					"testvar": Variable{
+						Type: TypeMap,
+						Value: map[string]Variable{
+							"key1": Variable{
+								Type: TypeString,
+								Value: "Hello",
+							},
+							"key2": Variable{
+								Type: TypeString,
+								Value: "World",
+							},
+						},
+					},
+				},
+			},
+			ShouldError: true,
+			ReturnType: TypeInvalid,
+		},
+		{
+			Name: "Multiple list expressions",
+			Output: &Output{
+				Exprs: []Node{
+					&VariableAccess{
+						Name: "testvar",
+					},
+					&VariableAccess{
+						Name: "testvar",
+					},
+				},
+			},
+			Scope: &BasicScope{
+				VarMap: map[string]Variable{
+					"testvar": Variable{
+						Type: TypeList,
+						Value: []Variable{
+							Variable{
+								Type: TypeString,
+								Value: "Hello",
+							},
+							Variable{
+								Type: TypeString,
+								Value: "World",
+							},
+						},
+					},
+				},
+			},
+			ShouldError: true,
+			ReturnType: TypeInvalid,
+		},
+		{
+			Name: "Multiple string expressions",
+			Output: &Output{
+				Exprs: []Node{
+					&VariableAccess{
+						Name: "testvar",
+					},
+					&VariableAccess{
+						Name: "testvar",
+					},
+				},
+			},
+			Scope: &BasicScope{
+				VarMap: map[string]Variable{
+					"testvar": Variable{
+						Type: TypeString,
+						Value: "Hello",
+					},
+				},
+			},
+			ReturnType: TypeString,
+		},
```

Comment:
Continuing comment from above double checking my understanding. This above case is testing `${var.foo}${var.foo}` where `foo` is a string set to `"Hello"` so this is valid and results in `HelloHello`. Correct?